### DISCUSSION TITLE
Add optional support for compiler sanitizers

### DISCUSSION THREAD

**Discussion by fbusato** [NVIDIA/cccl#2240] [General] [answered]
Sanitizers are extremely useful for detecting hard-to-find and subtle bugs. They include UB sanitizer, address sanitizer, memory sanitizer, and thread sanitizer. They can work in conjunction with CUDA code with a few workarounds. They are complementary to CUDA specific sanitizers, `memcheck`, `synccheck`, `racecheck`, `initcheck`. I don't see such support for the current build configurations. It would be great to add them to make the CCCL libraries even more robust.

**Comment by jrhemstad** [comment, marked as answer]
fyi, we're tracking this work here: https://github.com/NVIDIA/cccl/issues/1618

**Reply by fbusato** [reply]
My understanding is that https://github.com/NVIDIA/cccl/issues/1618 is related to `compute-sanitizer` for device code, not host code.

**Reply by jrhemstad** [reply]
Hm, I hadn't considered that distinction when we made the issue. I'd be happy to expand it or add another issue for doing host-side sanitizers too.

**Reply by fbusato** [reply]
ok, I will go ahead and create a new issue. thanks
